mergify_engine/duplicate_pull.py

```diff
@@ -27,6 +28,11 @@ class DuplicateNeedRetry(Exception):
     pass
 
 
+@attr.s
```

Comment:
In python >= 3.7, you can use builtin https://docs.python.org/3/library/dataclasses.html